File: Java/백준/Silver/11536. 줄 세우기/줄 세우기.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int num = sc.nextInt();
        List<String> names = new ArrayList<>();
        for(int i=0; i<num; i++){
            names.add(sc.next());
        }
        int result = 0;
        for(int i=0; i<num-1; i++){
            int compare = names.get(i).compareTo(names.get(i+1));
            if(compare > 0){
                result--;
            }else if(compare < 0){
                result++;
            }
        }
        if(result == num-1){
            System.out.println("INCREASING");
        }else if(result + (num-1) == 0){
            System.out.println("DECREASING");
        }else{
            System.out.println("NEITHER");
        }

    }
}

File: Python/백준/Silver/11536. 줄 세우기/줄 세우기.py
num = int(input())
names = []

for _ in range(num):
    names.append(input())

def compare(name1, name2):
    if(name1 < name2): # name1이 더 앞인 경우
        return True
    else: # name2가 더 앞인 경우
        return False

result = 0
for i in range(num-1):
    cmp = compare(names[i], names[i+1])
    if(cmp):
        result -= 1
    else:
        result += 1
    
if(result + (num-1) == 0):
    print('INCREASING')
elif(result == (num-1)):
    print('DECREASING')
else:
    print('NEITHER')


